clients/mobile/src/authentication/register/hooks/useRegister.ts

```diff
@@ -69,17 +74,39 @@ export const useRegister = (): any => {
 
   const handleGoogleSignIn = async(): Promise<void> => {
     try {
-      await googleLogin()
-      navigation.navigate(SCREENS.PROFILE)
+      const isGoogleSignedIn = await googleLogin()
+      if (isGoogleSignedIn) {
+        auth?.setIsAuthenticated(true)
+        registerUserDeviceForNotification(fetchClient)
+        navigation.dispatch(
+          CommonActions.reset({
+            index: 0,
+            routes: [{ name: SCREENS.ADD_PERSONAL_INFO }]
+          })
+        )
+      } else {
+        setSocialLoginError('Google login failed. Please try again.')
+      }
     } catch (error) {
       setSocialLoginError('Failed to sign in with Google.')
     }
   }
 
   const handleFacebookSignIn = async(): Promise<void> => {
     try {
-      await facebookLogin()
-      navigation.navigate(SCREENS.PROFILE)
+      const isFacebookSignedIn = await facebookLogin()
+      if (isFacebookSignedIn) {
+        auth?.setIsAuthenticated(true)
```

Comment:
same

Reply:
Remove the chaining operator. [Commit](https://github.com/CraftsmenLtd/BloodConnect/pull/239/commits/4356c490c82c28f3876f0b0d21519b1caaff1826)

```
const { setIsAuthenticated } = useAuth()
setIsAuthenticated(true)
```